src/test/groovy/com/intuit/graphql/orchestrator/integration/QueryDirectiveSpec.groovy

```diff
@@ -1,59 +1,241 @@
+
+        def userResponse = [
+                data: [
+                        users: [
+                                [id: "user-1", firstName: "Delilah", lastName: "Hadfield" ],
+                                [id: "user-2", firstName: "Huong", lastName: "Seamon" ],
+                                [id: "user-3", firstName: "Geraldine", lastName: "Gower" ]
+                        ]
+                ]
+        ]
+        SimpleMockServiceProvider userService = createSimpleMockService("userService", userSchema, userResponse)
+
+        ServiceProvider[] services = [ bookService, petsService, userService ]
+        specUnderTest = createGraphQLOrchestrator(services)
 
-        assert orchestrator.getSchema().isSupportingMutations()
+        assert specUnderTest.getSchema().isSupportingMutations()
 
         when:
-        // Test query using ExecutionInput
         ExecutionInput booksAndPetsEI = ExecutionInput.newExecutionInput().query('''
-            query BooksPetsAndUsers($includeType: Boolean!) { 
-                books { 
-                    id name 
+            query BooksPetsAndUsers($includeType: Boolean!) {
+                books {
+                    id name
                 }
-                pets { 
+                pets {
                     id name type @include(if: $includeType)
                 }
-                users { 
+                users {
                     id firstName lastName @include(if: $includeType)
-                } 
+                }
             }
         ''')
                 .variables(ImmutableMap.of("includeType", Boolean.TRUE))
                 .build()
 
-        Map<String, Object> executionResult = orchestrator.execute(booksAndPetsEI).get()
-                .toSpecification()
+        Map<String, Object> executionResult = specUnderTest.execute(booksAndPetsEI).get().toSpecification()
 
         then:
         executionResult.get("errors") == null
         executionResult.get("data") != null
 
         Map<String, Object> dataValue = (Map<String, Object>) executionResult.get("data")
         dataValue.keySet().containsAll("pets", "books", "users")
-        ((List<Map<String, Objects>>) dataValue.get("pets")).size() == 3
         ((List<Map<String, Objects>>) dataValue.get("books")).size() == 3
+        ((List<Map<String, Objects>>) dataValue.get("pets")).size() == 3
         ((List<Map<String, Objects>>) dataValue.get("users")).size() == 3
+
+        //  petsService
+        ExecutionInput petsServiceExecutionInput = getCapturedDownstreamExecutionInput(petsService)
+        Map<String, Object> petsServiceVariables = petsServiceExecutionInput.getVariables()
+        petsServiceVariables.size() == 1
+        petsServiceVariables["includeType"] == true
+
+        Document petsServiceDocument = parser.parseDocument(petsServiceExecutionInput.getQuery())
+        OperationDefinition petsServiceOperationDefinition = petsServiceDocument.getOperationDefinition("BooksPetsAndUsers").get()
```

Comment:
It's good we capture these objects. I think we need to validate that the include directive is being passed to downstream.  We can add this code:

```
        Field pet_type_field = getFieldWithPath("pets.type", petsServiceOperationDefinition.getSelectionSet())
        pet_type_field.getDirectives().size() == 1
        pet_type_field.getDirectives().get(0).getName() == "include"
        pet_type_field.getDirectives().get(0).getArgument("if") != null
```